### DISCUSSION TITLE
PlexCleaner and ffmpeg/handbrake variables?

### DISCUSSION THREAD

**Discussion by jrosetto** [ptr727/PlexCleaner#138] [Q&A] [answered]
Running a physical server 2019 with a video card

Is there a way I can modify the parameters being passed when converting videos?

I have an nvidia card that can decode/encode way faster and cheaper than a CPU and would like to use that.

I have FFMPEG working with the card when I manually pass the commands so I should be able to accomplish the same with PlexCleaner.

Thanks

**Comment by ptr727** [comment]
There is not currently a way to configure custom parameters for encoding.
I could add custom config parameters for ffmpeg and handbrake encoding steps.

1) What are the parameters you use for ffmpeg?
2) Can you try to get hardware encoding working with handbrake cli, as ffmpeg or handbrake are used for encoding depending on media attributes, e.g. failed ffmpeg or deinterlacing using using handbrake.

**Reply by jrosetto** [reply]
For my card the '-hwaccel cuda' seems to work great for ffmpeg.  Depending on the card I believe this may work as well '-c:v h264_cuvid'.
I believe ffmpeg supports AMD cards as well but don't have one to test.

Looking up the parameters for handbrakecli.exe it looks like this will do it but I will test to verify when I have time.  Might not be until next week as I have a busy weekend ahead of me.
--encopts="gpu=any"

**Reply by ptr727** [reply]
I was looking at the code, and it is not so easy to change the values as the commandline is dynamically generated, and if I'm going to support hardware encoding I'd like it to be more generic to support Intel QuickSync and NVidia options.

I'd suggest for now you fork the project and modify the code to change the codec type and global options in FfMpegTool.cs.
See: https://developer.nvidia.com/blog/nvidia-ffmpeg-transcoding-guide/

E.g.
```
    private const string H264Codec = "h264_cuvid";
    private const string H265Codec = "hevc_cuvid";
    private const string Snippet = "-ss 0 -t 180";
    private const string GlobalOptions = "-analyzeduration 2147483647 -probesize 2147483647 -hwaccel cuda -hwaccel_output_format cuda";
```

**Reply by ptr727** [reply, marked as answer]
Check out [v3 beta](https://github.com/ptr727/PlexCleaner/discussions/140), I'd like your testing input please.
